### ISSUE TITLE
How to uninstall ExplorerPatcher after virus detection?

### ISSUE THREAD

**Issue by Solid-Bob** [valinet/ExplorerPatcher#3682]
I installed version [22621.3880.66.5](https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3880.66.5_5094108), because the program recommended it. It downloaded as `C:\Users\L\AppData\Roaming\ExplorerPatcher\Update for ExplorerPatcher from https꞉∕∕github.com∕valinet∕ExplorerPatcher∕releases∕latest∕download∕ep_setup.exe`. The file name was this long. The `:` and `/` was replaced with a similarly looking Unicode character.

While installing, Avast antivirus detected "Win64:MalwareX-gen [Trj]" in it. I created an exception, and installed anyway. After that I deleted the file. Shortly after Avast insisted that I restart the computer, because this is the only way it can quarantine the file. It was referring to the file I deleted a minute ago, nevertheless I restarted.

Windows restarted with the default Windows 11 task bar. ExplorerPatcher was not working any more. `C:\Program Files\ExplorerPatcher\ep_setup.exe` disappeared too, therefore I cannot uninstall it from *Windows Settings* / *Apps* / *Installed apps* (it needs `ep_setup.exe`). Later I found in antivirus logs, that the same threat was detected in this file too, so probably the antivirus deleted it.

Now, how do I uninstall ExplorerPatcher? I want to remove all files, folders, autostart and registry entries it added, and restore all files it modified to original, and remove ExplorerPatcher from the list of *Installed apps*.

* Can I uninstall manually? I didn't find any manual uninstall steps ([link](https://github.com/valinet/ExplorerPatcher/wiki/Installer-How-To#how-to-uninstall)).
* Or, was the `Update for ExplorerPatcher from https...ep_setup.exe` and the `C:\Program Files\...\ep_setup.exe` the same? In this case could I add an antivirus exception to these folders, download the latest `ep_setup.exe`, and run it with `/uninstall`? Would that work?

Note: You should have a *separate* `ep_uninstall.exe`, that is modified less often, is not detected as a virus for sure, and therefore works always.

**Comment by Epyon83** [comment]
Same here. Windows Defender went wild but ep_setup.exe sounds okay?

**Comment by Jazzzed** [comment]
Did you follow the instructions for setting up Windows Defender exclusions that are found here?
https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3880.66.5_5094108

I installed the latest EP version without any problems.

**Comment by Amrsatrio** [comment]
If you follow the instructions for excluding the paths properly, everything will run smoothly.

**Comment by Solid-Bob** [comment]
You used an incorrect tense. The proper way to say it is this: If I followed the instructions for excluding the paths properly, everything would have run smoothly. Unfortunately, I didn't set the exclusions in advance. So, now I am in trouble. The question is the same, as in my first post; please help.

**Comment by pyrates999** [comment]
see here for how to uninstall: https://github.com/valinet/ExplorerPatcher/wiki/Installer-How-To#how-to-uninstall

**Comment by Amrsatrio** [comment]
These are the folders that you need to exclude. Then download ep_setup to %APPDATA%\ExplorerPatcher or somewhere else excluded from scans, run it, and then uninstall again. Problem solved.

C:\Program Files\ExplorerPatcher
%APPDATA%\ExplorerPatcher
C:\Windows\dxgi.dll
C:\Windows\SystemApps\Microsoft.Windows.StartMenuExperienceHost_cw5n1h2txyewy
C:\Windows\SystemApps\ShellExperienceHost_cw5n1h2txyewy

**Comment by Solid-Bob** [comment]
So, you suggest installing it overwriting the broken installation, then uninstalling it, right?

I'm not blaming anyone currently. I just want to solve this problem.

**Comment by Jazzzed** [comment]
Yes your best bet is to create the proper exclusions in your Antivirus or Defender, install it in full, and then uninstall in full using one of the methods in the link that pyrates99 provided. Then remove the exclusions.

**Comment by pyrates999** [comment]
> I am actually tired dealing with all these, you know. Microsoft nuked the 10 start menu on this month's patch which pushed me into releasing 66.4+ into the release channel, superseeding the known good 65.5. 66.4 bundles the start menu files from the latest 226xx build that had it.
> 
> It was either leaving people stranded on 65.5 so that they lose the 10 start menu, or just watching the world burn with the faulty 66.4 and 66.5. No good option really. We all should blame Microsoft for everything that happened.

Maybe it's time to look into reimplementing the windows 10 start menu the way you reimplemented the windows 10 taskbar.

**Comment by Solid-Bob** [comment]
I successfully uninstalled ExplorerPatcher. It is no longer in _Windows Settings_ / _Apps_ / _Installed apps_.

Can I delete the following paths and their descendants from the registry?
* `HKEY_CURRENT_USER\Software\ExplorerPatcher`
* `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\ExplorerPatcher`

**Comment by pyrates999** [comment]
yes
